game/zx_spectrum_tap.ksy

```diff
@@ -0,0 +1,52 @@
+meta:
+  id: zx_spectrum_tap
+  file-extension: tap
+  endian: le
+  license: CC0-1.0
+  title: ZX Spectrum tape file format
+  xref:
+    justsolve: TAP_(ZX_Spectrum)
+    pronom: fmt/801
+doc-ref: https://faqwiki.zxnet.co.uk/wiki/TAP_format
+seq:
+  - id: block
+    type: block
+    repeat: eos
+enums:
+  flag_enum:
+    0x00: header
+    0xFF: data
+  header_type_enum:
+    0: program
+    1: num_array
+    2: char_array
+    3: bytes
+types:
+  block:
+    seq:
+      - id: len_header
+        contents: [0x13, 0x00]
+      - id: flag
+        type: u1
+        enum: flag_enum
+      - id: header
+        type: header_block
+      - id: data
+        size: header.len_data + 4
+  header_block:
+    seq:
+      - id: header_type
+        type: u1
+        enum: header_type_enum
+      - id: filename
+        size: 10
+        pad-right: 0x20
+      - id: len_data
+        type: u2
+      - id: param1
+        type: u2
+      - id: param2
```

Comment:
How about `doc`? And why not
```yaml
- id: params
  type: u2
  repeat: expr
  repeat-expr: 2
```
?

Reply:
That's certainly an option. I'll review.

Reply:
Not sure if it's a good idea to make this an array. Semantically, the two "param" values are not an array, they are simply two `u2` values that have nothing to do with each other.

By the way, the meaning of the "param" values is known, but it depends on the `header_type`. It would be possible to create individual structures for each interpretation of the "param" values and select the correct one using a `switch-on` based on `header_type`.

Reply:
Agreed. Working on having separate param structures - just having trouble trying to avoid the `body_*` scenario described in the documentation.

Reply:
Done. Would appreciate a review.

Reply:
Much better, thanks.